src/borg/repository.py

```diff
@@ -190,6 +190,9 @@ def load_key(self):
         return keydata.encode('utf-8')  # remote repo: msgpack issue #99, returning bytes
 
     def get_free_nonce(self):
+        if not self.lock.got_exclusive_lock():
+            raise AssertionError("bug in code, exclusive lock should exist here")
```

Comment:
raise is the proper way to go here imho, it's not a casual "oh btw." assert, but something we wouldn't want to lose just because someone (user, packager, admin, ...) thought that -O globally is a good idea to improve performance by 0.1 % ;D